master/internal/user/service.go

```diff
@@ -107,6 +107,30 @@ func (s *Service) ProcessAuthentication(next echo.HandlerFunc) echo.HandlerFunc
 	}
 }
 
+// ProcessProxyAuthentication is a middleware processing function that attempts
+// to authenticate incoming HTTP requests coming through proxies.
+func (s *Service) ProcessProxyAuthentication(c echo.Context) (done bool, err error) {
+	token, err := s.extractToken(c)
+	if err != nil {
+		return true, c.Redirect(
+			http.StatusSeeOther,
+			fmt.Sprintf("/det/login?redirect=%s", c.Request().URL.Path),
```

Comment:
This throws away query parameters. Probably not that big a deal for the current use cases, but probably something to be avoided. Just dropping `.Path` seems to preserve the query, though there might still be encoding issues or something, I don't know.